### DISCUSSION TITLE
using throw from within then($onRejected)

### DISCUSSION THREAD

**Discussion by quazardous** [reactphp/reactphp#479] [Q&A] [answered]
I'm trying to handle errors with `$reactBrowser->requestStreaming($method, $url, $headers, $body)->then()`

```php
$reactBrowser->requestStreaming($method, $url, $headers, $body)->then(
    function (ResponseInterface $response) {},
    function (\Exception $reason) {
            echo "DEAD !!!\n";
            throw $reason; // <<< It's trapped somewhere
   });
$loop->run();
```

I'm throwing the `$reason` but it's trapped somewhere.

Is it a bug or is it by design ?

**Comment by SimonFrings** [comment, marked as answer]
Hey @quazardous, this is not a bug and a known behavior, you can read more about it in https://github.com/reactphp/promise/issues/87. You can try to handle the exception by using `try` `catch`, but we're aware that this is not the best behavior. As we speak we're in the middle of preparing a new major version (v3), where we also want to introduce better error and exception handling, @clue wrote about this in our [wish list ticket](https://github.com/orgs/reactphp/discussions/472) here on GitHub : https://github.com/orgs/reactphp/discussions/472#discussioncomment-3880271

I hope this helps :+1:

**Reply by quazardous** [reply]
thx for the answer !
